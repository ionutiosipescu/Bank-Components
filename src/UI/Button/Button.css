.btn {
  height: 30px;
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: 50px;
  border: 1px solid var(--purple);
  background: var(--white);
  color: var(--purple);
  transition: 0.3s;
  margin: 10px 0 10px 0;
  cursor: pointer;
}

/* Size */

.btn-sm {
  width: 100px;
}

.btn-md {
  width: 150px;
}

.btn-lg {
  width: 250px;
  height: 40px;
  font-size: 1rem;
}

.btn-100 {
  width: 100%;
  font-size: 1rem;
}

.btn-fit {
  width: fit-content;
  padding: 0 10px 0 10px;
}

.btn-circle {
  width: 50px;
  height: 50px;
  border-radius: 50%;
}

/* Type */

.btn-primary {
  background: var(--purple);
  color: var(--white);
}

.btn-secondary {
  background: var(--white);
  color: var(--purple);
}

/* Hover */

.btn:hover {
  background: var(--blue);
  border: 1px solid var(--blue);
  color: var(--white);
}
