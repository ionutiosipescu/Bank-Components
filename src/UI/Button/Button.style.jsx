import styled from "styled-components";
import { sizeBtn } from "../../../utils/componentSizes/buttonSizes";

export const Btn = styled.button`
  z-index: 1;
  display: flex;
  justify-content: center;
  align-items: center;
  border: 1px solid var(--purple);
  transition: 0.3s;
  margin-top: 10px;
  margin-bottom: 10px;
  border-radius: 50px;
  box-shadow: 0px 0px 12px -8px rgba(21, 20, 47, 0.8);

  ${(props) =>
    props.size === "sm"
      ? sizeBtn.sm
      : props.size === "md"
      ? sizeBtn.md
      : props.size === "lg"
      ? sizeBtn.lg
      : props.size === "xl"
      ? sizeBtn.xl
      : props.size === "round"
      ? sizeBtn.round
      : sizeBtn.fullWidth};
  ${(props) =>
    props.primary === "primary"
      ? "background: var(--purple); color: var(--text-light);"
      : "background: var(--white); color: var(--purple);"}
  ${(props) =>
    props.disabled ? "opacity: 0.5; cursor: default;" : "cursor: pointer;"}
  &:hover {
    scale: 1.08;
    ${(props) =>
      props.primary === "primary"
        ? "background: var(--purple-dark); color: var(--text-light);"
        : "background: var(--purple-light); color: var(--text-light); border: 1px solid var(--purple-dark);"}
  }
  & ${(props) => props.children} {
    transition: 0.3;
  }
`;
