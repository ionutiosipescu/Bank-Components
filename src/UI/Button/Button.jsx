import React from "react";
import { PropTypes } from "prop-types";
import { Btn } from "./Button.style";

function Button({ label, size, primary, handleClick, children, ...props }) {
  return (
    <Btn
      size={size}
      onClick={handleClick}
      primary={primary}
      id="button-global"
      {...props}
    >
      {label}
      {children}
    </Btn>
  );
}

Button.propTypes = {
  id: PropTypes.string,
  label: PropTypes.string,
  to: PropTypes.string,
  size: PropTypes.oneOf(["sm", "md", "lg", "xl", "round", "fullWidth"]),
  primary: PropTypes.oneOf(["primary", "secondary"]),
  handleClick: PropTypes.func,
};

export default Button;
