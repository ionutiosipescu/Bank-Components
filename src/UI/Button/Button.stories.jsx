import Button from "./Button";

export default {
  title: "Components/Button",
  component: Button,
  argTypes: { handleClick: { action: "handleClick" } },
};

const Template = (args) => <Button {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  label: "Click me",
  primary: "primary",
  size: "sm",
};

export const Secondary = Template.bind({});
Secondary.args = {
  label: "Click me",
  primary: "secondary",
  size: "sm",
};
