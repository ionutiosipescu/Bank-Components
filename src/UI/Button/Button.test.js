import React from "react";
import { shallow } from "enzyme";
import Button from "./Button";

describe("Button component", () => {
  it("renders without crashing", () => {
    const wrapper = shallow(<Button />);
    expect(wrapper.exists()).toBe(true);
  });

  it("renders with label prop", () => {
    const label = "Click me";
    const wrapper = shallow(<Button label={label} />);
    expect(wrapper.find(`[id="button-global"]`).text()).toEqual(label);
  });

  it("calls handleClick prop on click", () => {
    const handleClick = jest.fn();
    const wrapper = shallow(<Button handleClick={handleClick} />);
    wrapper.find(`[id="button-global"]`).simulate("click");
    expect(handleClick).toHaveBeenCalled();
  });

  it("matches snapshot", () => {
    const wrapper = shallow(
      <Button label="Click me" size="sm" primary="primary" />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
