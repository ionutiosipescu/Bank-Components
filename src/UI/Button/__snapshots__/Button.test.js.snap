// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component matches snapshot 1`] = `
<styled.button
  id="button-global"
  primary="primary"
  size="sm"
>
  Click me
</styled.button>
`;
